src/features/import/hooks/useUIDataColumns.tsx

```diff
@@ -0,0 +1,433 @@
+import messageIds from '../l10n/messageIds';
+import notEmpty from 'utils/notEmpty';
+import { updateColumn } from '../store';
+import useTags from 'features/tags/hooks/useTags';
+import { ZetkinTag } from 'utils/types/zetkin';
+import { CellData, Column, ColumnKind } from '../utils/types';
+import { Msg, useMessages } from 'core/i18n';
+import { useAppDispatch, useAppSelector } from 'core/hooks';
+
+export type UIDataColumn = {
+  assignTag: (tagId: number, value: CellData) => void;
+  columnValuesMessage: string;
+  deselectOrg: (value: CellData) => void;
+  getAssignedTags: (value: CellData) => ZetkinTag[];
+  getSelectedOrgId: (value: CellData) => number | null;
+  numRowsByUniqueValue: Record<string | number, number>;
+  numberOfEmptyRows: number;
+  originalColumn: Column;
```

Comment:
Maybe this could have a generic type (`CType`), and `UIDataColumn` could be have the signature `UIDataColumn<CType extends Column>`.